src/alloc.c

```diff
@@ -476,8 +476,6 @@ static jl_lambda_info_t *jl_clone_thunk(jl_lambda_info_t *linfo, jl_tupletype_t
     new_linfo->nargs = linfo->nargs;
     new_linfo->isva = linfo->isva;
     new_linfo->rettype = linfo->rettype;
-    new_linfo->def = NULL;
-    new_linfo->specTypes = types;
```

Comment:
Should this be left in or at least add a comment so user won't forget to initialize this?